src/rgw/rgw_rados.h

```diff
@@ -175,7 +175,9 @@ struct RGWObjState {
   bufferlist olh_tag;
   uint64_t pg_ver{false};
   uint32_t zone_short_id{0};
-
+  //off_t start{std::numeric_limits<off_t>::max()};
+  //off_t end{std::numeric_limits<off_t>::max()};
+  prefetch_range *range;
```

Comment:
this field should replace `bool prefetch_data`. a nullptr indicates `!prefetch_data`

since `RGWObjState` acts like a cache, it's probably not a good idea to store this as a pointer; we may end up caching a pointer to memory on the stack that goes away. `std::optional<prefetch_range>` may be a better fit